### PR TITLE
add f1 help keywords to pmc and pm ui

### PR DESCRIPTION
**Pull request by rohit21agrawal**
Fixes: https://github.com/NuGet/Home/issues/5912

Adds help keyword to redirect to the right docs page when F1 is pressed for a control.

Doc side PR : https://github.com/NuGet/docs.microsoft.com-nuget/pull/642

### CODE REVIEW

File: src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs [hunk 1]
```diff
@@ -480,6 +480,19 @@ private void PowerConsoleService_ExecuteEnd(object sender, EventArgs e)
                     string.Empty,
                     null,
                     out windowFrame);
+                if(windowFrame != null)
```
Comment (emgarten) [inline src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs]: format the doc

File: src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs [hunk 2]
```diff
@@ -480,6 +480,19 @@ private void PowerConsoleService_ExecuteEnd(object sender, EventArgs e)
...
+                    // Set F1 help keyword
+                    object varUserContext = null;
+                    if (ErrorHandler.Succeeded(windowFrame.GetProperty((int)__VSFPROPID.VSFPROPID_UserContext, out varUserContext)))
+                    {
+                        var userContext = varUserContext as IVsUserContext;
+                        if (userContext != null)
+                        {
+                            userContext.AddAttribute(VSUSERCONTEXTATTRIBUTEUSAGE.VSUC_Usage_LookupF1, "keyword", "VS.NuGet.PackageManager.UI");
+                        }
+                    }
```
Comment (emgarten) [inline src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs]: Can you make this into a helper method in visualstudio common?

**Review by emgarten — CHANGES_REQUESTED** [review]

**Comment by rohit21agrawal** [comment]
@emgarten  done

File: src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs [hunk 3]
```diff
@@ -147,6 +147,14 @@ public override void OnToolWindowCreated()
 
             // Register key bindings to use in the editor
             var windowFrame = (IVsWindowFrame)Frame;
+            if (windowFrame != null)
+            {
+                // Set F1 help keyword
+                if (windowFrame != null)
```
Comment (emgarten) [inline src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs]: remove duplicate check 😄

**Review by emgarten — APPROVED** [review]
Glad to see f1 support fixed! 🚀

File: src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs [hunk 4]
```diff
@@ -147,6 +147,14 @@ public override void OnToolWindowCreated()
...
+                {
+                    WindowFrameHelper.AddF1HelpKeyword(windowFrame, keywordValue: "VS.NuGet.PackageManager.Console");
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs]: nit: create constants for these hard code strings.

**Review by jainaashish — APPROVED** [review]
LGTM

**Review by mishra14 — APPROVED** [review]

**Comment by Newrad0603** [comment]
LGTM

**Review by Huy-Nguyen-MSFT — COMMENTED** [review]
Looks good.

**Pull request merged**